lib/media/segment_index.js

```diff
@@ -361,24 +361,25 @@ shaka.media.SegmentIndex = class {
   }
 
 
-  /** @return {!shaka.media.SegmentIterator} */
+  /** @return {?shaka.media.SegmentIterator} */
```

Comment:
@joeyparrish  I'm not completely certain what you are proposing here. Can you you show me the line by line changes you think are necessary?

Reply:
Something like this:

```js
  /** @return {!shaka.media.SegmentIterator} */
  [Symbol.iterator]() {
    const iter = this.getIteratorForTime(0);
    goog.asserts.assert(iter != null, 'Iterator for 0 should never be null!');
    return iter;
  }
```

We keep the return signature as "non-nullable iterator", and we assert at runtime that the returned iterator for 0 is not null.  (It should never be, as it should always point to the first reference.)

The assertion convinces the compiler that even though `getIteratorForTime()` could return null, it can treat the return value here as non-null.  If that assertion ever fails at runtime in the demo app, it will show up in the dev console in debug mode.  If that assertion ever fails during an automated test, it will fail the test case.